### DISCUSSION TITLE
Do we have a spatial gradient in material?

### DISCUSSION THREAD

**Discussion by Oops-Qiao** [idaholab/moose#22259] [Q&A General] [answered]
Hi, 

I can see there is adCoupledVectorGradient for coupled variable. Do we also have something similar when we want to retrieve the spatial gradients of some properties from material?
Thank you.

Regards,
Qiao

**Comment by GiudGiud** [comment, marked as answer]
Not that I know of but you can easily compute it and define it yourself. 

If you’re just running through the tutorial and need some guidance no need to post in several places by the way

**Reply by Oops-Qiao** [reply]
Ok, I see. thanks
